receiver/googlecloudpubsubreceiver/README.md

```diff
@@ -12,18 +12,22 @@ This receiver gets OTLP messages from a Google Cloud [Pubsub](https://cloud.goog
 
 The following configuration options are supported:
 
+* `mode` (Optional): The mode that the receiver is in, either `push` or `pull` is supported. The default is `pull`.
 * `project` (Optional): The Google Cloud Project of the client connects to.
-* `subscription` (Required): The subscription name to receive OTLP data from. The subscription name  should be a 
+* `subscription` (Required in pull mode): The subscription name to receive OTLP data from. The subscription name  should be a 
   fully qualified resource name (eg: `projects/otel-project/subscriptions/otlp`).
 * `encoding` (Optional): The encoding that will be used to received data from the subscription. This can either be
   `otlp_proto_trace`, `otlp_proto_metric`, `otlp_proto_log`, or `raw_text` (see `encoding`).  This will only be used as 
   a fallback, when no `content-type` attribute is present.
 * `compression` (Optional): The compression that will be used on received data from the subscription. When set it can 
   only be `gzip`. This will only be used as a fallback, when no `content-encoding` attribute is present.
-* `endpoint` (Optional): Override the default Pubsub Endpoint, useful when connecting to the PubSub emulator instance
-  or switching between [global and regional service endpoints](https://cloud.google.com/pubsub/docs/reference/service_apis_overview#service_endpoints).
+* `endpoint` (Optional): In `pull` mode, override the default Pubsub Endpoint, useful when connecting to the PubSub emulator instance
+  or switching between [global and regional service endpoints](https://cloud.google.com/pubsub/docs/reference/service_apis_overview#service_endpoints). In `push` mode the endpoint is that address that the http server listens to for receiving Pub/Sub messages. 
```

Comment:
Isn't the `endpoint` option only used for `pull` mode? It looks like the `push` mode uses` push.endpoint` and not `endpoint`.

Reply:
You're right; I refactored it after I wrote the documentation. Fixed.